src/lib.rs

```diff
@@ -321,347 +316,491 @@ impl<'de, E: Curve> serde::Deserialize<'de> for ExtendedKeyPair<E> {
     }
 }
 
-/// Marker for a curve supported by SLIP10 specs and this library
-///
-/// Only implement this trait for the curves that are supported by SLIP10 specs.
-/// Curves provided by the crate out-of-box in [supported_curves] module already
-/// implement this trait.
-pub trait SupportedCurve {
-    /// Specifies which curve it is
-    const CURVE_TYPE: CurveType;
-}
-#[cfg(feature = "curve-secp256k1")]
-impl SupportedCurve for supported_curves::Secp256k1 {
-    const CURVE_TYPE: CurveType = CurveType::Secp256k1;
+/// HD derivation
+pub trait HdWallet<E: Curve>: DeriveShift<E> {
+    /// Derives child extended public key from parent extended public key
+    ///
+    /// Returns error if child key is not defined for given child index
```

Comment:
How does it return an error? It's type is literally errorless.

Ditto for the other 4 functions with this comment here

Reply:
Ah yes, I forgot to remove that

Reply:
I was experimenting, I added bip32 support (it may return an error), but then decided that it's not worth the hassle. I'll update the comments